### PR TITLE
Adding topology option to the overlay NAD

### PR DESCRIPTION
**Pull request by Anatw**
##### Short description:
Adding topology option to the overlay NAD resource.

### CODE REVIEW

File: ocp_resources/network_attachment_definition.py [hunk 1]
```diff
@@ -201,6 +201,7 @@ class OVNOverlayNetworkAttachmentDefinition(NetworkAttachmentDefinition):
     def __init__(
         self,
         network_name=None,
+        overlay_topology=None,
```
Comment (myakove) [inline ocp_resources/network_attachment_definition.py:204]: The name in the resource is `topology`, please keep it.

File: ocp_resources/network_attachment_definition.py [hunk 2]
```diff
@@ -218,14 +219,17 @@ def __init__(
...
         self.network_name = network_name
+        self.overlay_topology = overlay_topology
 
     def to_dict(self):
         super().to_dict()
         if not self.yaml_file:
             if not self.network_name:
                 raise ValueError("network_name is required")
+            if not self.overlay_topology:
+                raise ValueError("overlay_topology is required")
```
Comment (myakove) [inline ocp_resources/network_attachment_definition.py:230]: Is topology required? I didn't find anything about it in the docs.
Also it wasn't required till now, what changed?

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/network_attachment_definition.py (hunk 2 above)
Comment (Anatw) [reply ocp_resources/network_attachment_definition.py:230]: Before, we had only one type of topology - "layer2". Now we also have a "localnet" topology.
https://docs.openshift.com/container-platform/4.14/networking/multiple_networks/configuring-additional-network.html#configuration-ovnk-additional-networks_configuring-additional-network

**Review by rnetser — CHANGES_REQUESTED** [review]
Please update api doc link

**Review by myakove — APPROVED** [review]

File: ocp_resources/network_attachment_definition.py [hunk 3]
```diff
@@ -201,13 +201,14 @@ class OVNOverlayNetworkAttachmentDefinition(NetworkAttachmentDefinition):
     def __init__(
         self,
         network_name=None,
+        topology=None,
```
Comment (rnetser) [inline ocp_resources/network_attachment_definition.py:204]: please add to args

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/network_attachment_definition.py (hunk 3 above)
Comment (myakove) [reply ocp_resources/network_attachment_definition.py:204]: docstring

Comment (Anatw) [reply ocp_resources/network_attachment_definition.py:204]: Done

File: ocp_resources/network_attachment_definition.py [hunk 4]
```diff
@@ -201,31 +201,36 @@ class OVNOverlayNetworkAttachmentDefinition(NetworkAttachmentDefinition):
...
         Create and manage an OVN k8s overlay NetworkAttachmentDefinition (a switched, layer 2, topology network).
 
         API reference:
-        https://docs.openshift.com/container-platform/4.13/networking/multiple_networks/configuring-additional-network.html#configuration-ovnk-network-plugin-json-object_configuring-additional-network
+        https://docs.openshift.com/container-platform/4.14/networking/multiple_networks/configuring-additional-network.html#configuration-ovnk-additional-networks_configuring-additional-network
 
         Args:
             network_name (str): The name of the network. This field is mandatory when not using yaml
                 file.
+            topology (str): The secondary network topology to be created - "layer2" or "localnet".
```
Comment (myakove) [inline ocp_resources/network_attachment_definition.py:216]: `"layer2" or "localnet".` please remove, we do not want to update this once we have another type.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/network_attachment_definition.py (hunk 4 above)
Comment (Anatw) [reply ocp_resources/network_attachment_definition.py:216]: Ack

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Comment by Anatw** [comment]
!cherry-pick v4.15

**Pull request merged**